the_aesir_escape_valhalla.txt

```diff
@@ -0,0 +1,9 @@
+Name:The Aesir Escape Valhalla
+ManaCost:2 G
+Types:Enchantment Saga
+K:Chapter:3:I,II,III
+SVar:I:DB$ ChangeZone | ChangeNum$ 1 | ChangeType$ Permanent.YouOwn | Mandatory$ True | Origin$ Graveyard | Destination$ Exile | SubAbility$ DBGainLife | SpellDescription$ Exile a permanent card from your graveyard. You gain life equal to its mana value.
+SVar:DBGainLife:DB$ GainLife | Defined$ You | LifeAmount$ ExiledWith$CardManaCost
+SVar:II:DB$ PutCounter | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select target creature you control | CounterType$ P1P1 | CounterNum$ ExiledWith$CardManaCost | SpellDescription$ Put a number of +1/+1 counters on target creature you control equal to the mana value of the exiled card.
+SVar:III:DB$ ChangeZone | ChangeType$ Card.StrictlySelf,ExiledWith | Origin$ Battlefield,Exile | Destination$ Hand | SpellDescription$ Return The Aesir Escape Valhalla and the exiled card to their owner's hand.

```

Comment:
was that ChangeType tested?

Reply:
No, I just asked on Discord if it would work.

Reply:
`Card.ExiledWithSource`